extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
@gianm isn't this what would actually happen?
...
6. close thread closes a `consumerConnector` from a previous loop of the executor
7. executor thread creates a new consumerConnector
8. the check for `executorService.isShutdown()` returns true and we break from the loop
9. the finally block executes and the new consumerConnector created in 7 is closed